docs/enhancements/scale-mode.md

```diff
@@ -0,0 +1,128 @@
+OSD has a massive investment in AppSRE and SRE-P to handle this.
+
+Furthermore, OSD hosts these hub shards in the public cloud.
+Certain edge scenarios may not have this luxury.
+For example, telcos may manage their cattle from baremetal servers under a cell tower.
+The cost of scaling out to multiple BM servers is significantly higher than that of growing or adding hub clusters in a cloud.
+
+## Proposal
+To get around etcd limits, we propose an architecture whereby as many objects as possible are spread across multiple "data planes" outside of the hub's etcd; and all workloads (containers) are run in the hub "control plane".
+
+**Data Plane:** A virtualized kube API server with its object database isolated from that of the "control plane".
+It might be an external OCP cluster.
+It might be a database service running on the hub.
+It might be $other.
+Basically it can be anything that
+- Can be connected to via a client built from a kubeconfig
+- Responds to k8s CRUD primitives (get, list, watch, create, patch, update, delete)
+
+Importantly, we must assume a data plane _can not_ run workloads (containers).
+
+**Control Plane:** The master "hub" wherein
+- The consuming software will run, exposing a UI to the entire fleet of spoke clusters.
+- All hive controllers will run.
+- Hive will run all "runnables" -- e.g. Job => Pod => Container => install-manager to provision a cluster.
+
+## Overview
+hive-operator will learn how to deploy the controller stack (hive-controllers, hiveadmission, hive-clustersync) to more than one target namespace.
+In each such namespace it will expect to find a secret containing a kubeconfig providing access to a data plane.
+(The data plane must be globally unique to preclude separate controller stacks colliding attempting to reconcile the same object.)
+For each k8s CRUD operation hive needs to perform, the controller stack will decide whether the operation involves a workload (container) or not.
+Operations involving workloads will be performed in the control plane, using the client based on the controller manager, as normal.
+Other operations will be performed in the data plane, using a client constructed from the aforementioned kubeconfig.
+
+## HiveConfig
+HiveConfig will grow one additional field, `Spec.ScaleMode`, a boolean.
+When false, hive will operate in normal mode, managing and deploying to the (single) namespace indicated by `Spec.TargetNamespace` (which defaults to `hive` if unspecified).
+When true, hive will ignore `Spec.TargetNamespace` and operate as further described below.
+
+The existing `Ready` condition in the `Status.Conditions` list will continue to be used to indicate the health of the deployments.
+If *any* data plane is unhealthy, the condition's `Status` will be set to `False` and its `Message` will enumerate the unhealthy namespaces.
```

Comment:
As I started looking into the code around this, I realized it wasn't going to work very well. Rewriting and adding this option to Alternatives (with others).

Reply:
✓